### PR TITLE
Add new domain status route for the all my sites domains view

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* Add new route for domain status screen that will be used for the all my sites domains view
* Fix the back button too

#### Testing instructions

* Open /domains/manage
* Click on any domain - the left navigation menu shouldn't change
* Click the back button - you should go back to the all my sites domains view

Details: p99Zz8-1em-p2

### CODE REVIEW

**Comment by hambai** [comment]
@Automattic/team-calypso I've added you here because this PR is introducing a confusing type of routes where we want to both have a site URL part and `selectedSite` in the state but at the same time we don't want to show the site specific sidebar navigation. See more details in the p2 post - https://wp.me/p99Zz8-1em

**Comment by jsnajdr** [comment]
> we don't want to show the site specific sidebar navigation.

@hambai What do you want to show instead? If I understand the P2 correctly, the sidebar would be always showing the "All Sites" view, even though the content view is showing domain/domains for a specific selected site?

Every route handler is responsible for rendering the sidebar and can choose any version of it that fits its needs. See the `navigation` route handler in `my-sites/controller` and the navigation components in `my-sites/navigation`.

The sidebar components (site picker, current site, menu ...) that need it can get a new prop like `forceAllSitesView`. Then they can ignore the selected site in Redux and always treat `selectedSite` as `null`.

File: client/my-sites/sidebar/index.jsx [hunk 1]
```diff
@@ -851,9 +851,14 @@ export class MySitesSidebar extends Component {
 
 function mapStateToProps( state ) {
 	const currentUser = getCurrentUser( state );
-	const selectedSiteId = getSelectedSiteId( state );
+
+	const isAllDomainsView = isUnderDomainManagementAll( getCurrentRoute( state ) );
```
Comment (jsnajdr) [inline client/my-sites/sidebar/index.jsx:855]: This `MySitesSidebar` component is a generic one that can be used (potentially or really) in many different Calypso environments, like Jetpack Cloud or a management app for WP for Teams (aka P2).

But with this code, it "needs to know" about a very specific part of the Calypso app, namely the domain management section. Generic components shouldn't have knowledge about or dependencies on specific ones.

I think it's better to add support for a new prop -- `forceAllSitesView`, or `observeSelectedSite={ false }` -- and pass the value from above. That would be the code that wires the app view together, e.g., the `/me/domains` route handler.

Comment (hambai) [reply client/my-sites/sidebar/index.jsx:855]: Cool - that's why I wanted someone with more experience to look into this part of the code. I'll update the PR and fix it! 🙇

Comment (hambai) [reply client/my-sites/sidebar/index.jsx:855]: OKay - I looked into this but it seems that `MySitesSidebar` is not exactly a generic component - it actually heavily depends on the state and it has checks for atomic sites, domain only sites and etc. However I've fixed the `CurrentSite` component as it is a generic component.

**Review by eltongo — APPROVED** [review]
LGTM

**Pull request merged**